### PR TITLE
More artifacts that are needed to make the gghbb analysis work and other cosmetic things

### PR DESCRIPTION
**Pull request by lgray**
- tanh -> sinh for eta calculation
- effective area lookup from text file
- histograms you keep are now defined in finalize
- faster construction from pt/eta/phi/mass if available
- deal with TMath:: functions in JEC formula definitions
- jec, jer, jer-sf, jec-uncertainties operating OK
- factorized jet corrector and JME style tools
- JetTransformer class to apply corrections easily

To Investigate Further:
- Improve evaluation speed of jetmet corrections

### CODE REVIEW

File: fnal_column_analysis_tools/lookup_tools/jersf_lookup.py [hunk 1]
```diff
@@ -5,17 +5,18 @@
 from copy import deepcopy
 import numba
 
-from numpy import sqrt,log
+from numpy import sqrt,log,exp,abs
```
Comment (nsmith-) [inline fnal_column_analysis_tools/lookup_tools/jersf_lookup.py]: Why not use the namespace? `np.x`

File: fnal_column_analysis_tools/lookup_tools/jme_standard_function.py [hunk 2]
```diff
@@ -5,29 +5,60 @@
...
+    try:
+        val = float(fstr)
+        fstr = 'np.full_like(%s,%f)' % ( varlist[0], val )
+    except ValueError:
+        val = fstr
     lstr = "lambda %s: %s" % (",".join(varlist), fstr)
     func = eval(lstr)
-    nfunc = numba.njit(func)
-    return nfunc
+    #nfunc = numba.njit(func)
```
Comment (nsmith-) [inline fnal_column_analysis_tools/lookup_tools/jme_standard_function.py]: I think the meaning of this function has drifted quite a bit

**Comment by nsmith-** [comment]
Can we get a few docstrings in this stuff?

**Comment by lgray** [comment]
@nsmith- Yeah, that'll come for sure. I think first I'm going to spend some time implementing more of the analysis and see what else needs to come into this PR.

File: fnal_column_analysis_tools/lookup_tools/jersf_lookup.py (hunk 1 above)
Comment (lgray) [reply fnal_column_analysis_tools/lookup_tools/jersf_lookup.py]: less faffing about in eval and string manipulation, but yeah it's not as clean.

File: fnal_column_analysis_tools/lookup_tools/jme_standard_function.py (hunk 2 above)
Comment (lgray) [reply fnal_column_analysis_tools/lookup_tools/jme_standard_function.py]: quite a lot yes.

Comment (lgray) [reply fnal_column_analysis_tools/lookup_tools/jme_standard_function.py]: Changed name of function

**Comment by lgray** [comment]
closes #28
closes #23
closes #17

**Pull request merged**